spyder_kernels/console/kernel.py

```diff
@@ -372,6 +374,21 @@ def close_all_mpl_figures(self):
         except:
             pass
 
+    def is_special_kernel_valid(self):
+        """
+        Check if optional dependencies are available for special consoles.
+        """
+        try:
+            if os.environ.get('SPY_AUTOLOAD_PYLAB_O') == 'True':
+                import matplotlib
+            elif os.environ.get('SPY_SYMPY_O') == 'True':
+                import sympy
+            elif os.environ.get('SPY_RUN_CYTHON') == 'True':
+                import cython
+        except ImportError as e:
+            return to_text_string(e)
```

Comment:
Let's change this to

```python
except Exception as e:
    if os.environ.get('SPY_AUTOLOAD_PYLAB_O') == 'True':
        return u'matplotlib'
    elif os.environ.get('SPY_SYMPY_O') == 'True':
        return u'sympy'
    elif os.environ.get('SPY_RUN_CYTHON') == 'True':
        return 'cython'
```

On the Spyder side we'll use those names to build a generic error message.